core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,102 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import java.util.List;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedItem;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final Long lastEpisodeID;
+    private final boolean dontShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
+
+        FeedPreferences prefs = feed.getPreferences();
+        if (!prefs.getKeepUpdated() || !prefs.getShowEpisodeNotification()) {
+            dontShowNotification = true;
+            lastEpisodeID = null;
+            return;
+        }
+
+        List<FeedItem> feedItems = DBReader.getFeedItemList(feed);
```

Comment:
How about using `PodDbAdapter.getFeedCounters` here? If you make the setting an argument of the method, the code here can only load the number of new items for a specific feed instead of loading the whole feed and all its items. I know this is slightly different to what I wrote earlier but that way, it could be even more efficient ;)

Reply:
is this always 100% accurate? I thought this might fail with a huge amount of episodes, because AntennaPod might delete records of old episodes at some point.
(bc your one of the maintainers and im not, i'm creating a commit now which implements this. I will revert it in case this AntennaPod does delete old records)

Reply:
> AntennaPod might delete records of old episodes at some point.

AntennaPod does not delete old items. Not seeing all episodes only happens when you subscribe to a new podcast and the publishers delete old episodes. I think it should be okay because even if we did clean up, we would probably not delete new items.